tests/framework/test_time.py

```diff
@@ -1,19 +1,28 @@
+from __future__ import annotations
+
 import math
+from typing import Any, Generator
+from unittest.mock import MagicMock
 
 import numpy as np
 import pandas as pd
 import pytest
+import pytest_mock
+from layered_config_tree import LayeredConfigTree
 
 from tests.helpers import Listener, MockComponentA, MockComponentB, MockGenericComponent
-from vivarium.framework.engine import SimulationContext as SimulationContext_
+from vivarium.component import Component
+from vivarium.framework.engine import Builder, SimulationContext
 from vivarium.framework.event import Event
+from vivarium.framework.results.observer import Observer
 from vivarium.framework.time import SimulationClock, get_time_stamp
 from vivarium.framework.utilities import from_yearly
 from vivarium.framework.values import ValuesManager, rescale_post_processor
+from vivarium.types import ClockStepSize, ClockTime, NumberLike
 
 
 @pytest.fixture
-def builder(mocker):
+def builder(mocker: pytest_mock.MockFixture) -> Any:
```

Comment:
You can return MagicMock (likew you do at the bottom of this file)

Reply:
Says it is returning Any. I didnt understand the difference here either.